src/function-builder-detectors/netlify-lambda.js

```diff
@@ -13,12 +14,20 @@ const detectNetlifyLambda = async function ({ dependencies, devDependencies, scr
   for (const key in scripts) {
     const script = scripts[key]
 
-    const match = script.match(/netlify-lambda build.* (\S+)\s*$/)
-    if (match) {
-      const [, src] = match
-      settings.src = src
-      settings.npmScript = key
-      break
+    if (script.match(/netlify-lambda\s+build/)) {
+      // E.g. ["netlify-lambda", "build", "functions/folder"]
+      const match = minimist(script.split(' '))
```

Comment:
This isn't whitespace safe (e.g. a functions folder with a whitespace will break this), but the current approach isn't either and this works for more use case.
🚀